pymeasure/adapters/adapter.py

```diff
@@ -34,44 +37,112 @@ class Adapter:
     This class should only be inherited from.
 
     :param preprocess_reply: optional callable used to preprocess strings
-        received from the instrument. The callable returns the processed string.
+        received from the instrument. The callable returns the processed
+        string. This argument is deprecated.
     :param kwargs: all other keyword arguments are ignored.
     """
 
     def __init__(self, preprocess_reply=None, **kwargs):
         self.preprocess_reply = preprocess_reply
         self.connection = None
+        self.log = logging.Logger("Adapter")
+        self.log.addHandler(logging.NullHandler())
+        if preprocess_reply is not None:
+            warn(
+                "Do not use `Adapter.values`, but `Instrument.values` instead. "
+                "Preprocess_reply is a values parameter.",
+                FutureWarning)
 
     def __del__(self):
         """Close connection upon garbage collection of the device"""
-        if self.connection is not None:
+        self.close()
+
+    def close(self):
+        """Close the connection."""
+        try:
             self.connection.close()
+        except AttributeError:
+            pass
 
-    def write(self, command):
-        """ Writes a command to the instrument
+    # Directly called methods. DO NOT OVERRIDE IN SUBCLASS!
+    def write(self, command, **kwargs):
+        """
+        Write a string command to the instrument.
```

Comment:
I always start the summary line of a multiline docstring immediately after the quotes (same line), but I guess that's taste? Both are allowed by PEP257.

Reply:
I got used to starting in a new line, and forgot, that the other way is also allowed. Therefore I changed it sometimes, thinking it is wrong. Thanks for reminding me of that option.
I'll look into it, what is more common in pymeasure and adapt to that style. As I see the advantage of starting in the first line, I might prefer that (your) option.

Reply:
We could start adding pydocstyle checking to CI. There's a setting about exactly this, btw: "D212 | Multi-line docstring summary should start at the first line"